File: app/src/main/java/com/faciee/cti/valbastrelu/eticket/main/ETkAppPreferences.java
package com.faciee.cti.valbastrelu.eticket.main;

import android.content.SharedPreferences;

import static com.faciee.cti.valbastrelu.eticket.util.SharedPreferencesUtil.clearAll;
import static com.faciee.cti.valbastrelu.eticket.util.SharedPreferencesUtil.getString;
import static com.faciee.cti.valbastrelu.eticket.util.SharedPreferencesUtil.putValueInSharedPref;
import static com.faciee.cti.valbastrelu.eticket.util.SharedPreferencesUtil.removeKeys;

/**
 * Created by valbastrelu on 09-Apr-18.
 */
public class ETkAppPreferences implements IETkAppPreferences {
	
	private static final String EMAIL_KEY = "EMAIL";
	
	private SharedPreferences sharedPreferences;
	
	public ETkAppPreferences(SharedPreferences sharedPreferences){
		this.sharedPreferences = sharedPreferences;
	}
	
	void clearAllPrefs(){
		clearAll(sharedPreferences);
	}
	
	@Override
	public void cleanPreferences() {
		removeKeys(sharedPreferences, EMAIL_KEY);
	}
	
	@Override
	public String getCurrentEmail() {
		return getString(sharedPreferences, EMAIL_KEY);
	}
	
	@Override
	public void setCurrentEmail(String email) {
		putValueInSharedPref(sharedPreferences, EMAIL_KEY, email);
	}
}

File: app/src/main/java/com/faciee/cti/valbastrelu/eticket/base/ETkAppPreferences.kt
package com.faciee.cti.valbastrelu.eticket.base

import android.content.SharedPreferences
import com.faciee.cti.valbastrelu.eticket.extensions.clearAll
import com.faciee.cti.valbastrelu.eticket.extensions.getSafeBoolean
import com.faciee.cti.valbastrelu.eticket.extensions.getSafeString
import com.faciee.cti.valbastrelu.eticket.extensions.putValueAndApply
import com.faciee.cti.valbastrelu.eticket.extensions.removeKeys

/**
 * Created by valbastrelu on 09-Apr-18.
 */
class ETkAppPreferences(private val sharedPreferences: SharedPreferences) {
	fun clearAllPrefs() {
		sharedPreferences.clearAll()
	}

	fun cleanPreferences() {
		sharedPreferences.removeKeys(EMAIL_KEY)
	}

	var currentEmail: String
		set(value) = sharedPreferences.putValueAndApply(EMAIL_KEY, value)
		get() = sharedPreferences.getSafeString(EMAIL_KEY)

	var scheduleCache: Boolean
		set(value) = sharedPreferences.putValueAndApply(SCHEDULE_CACHE, value)
		get() = sharedPreferences.getSafeBoolean(SCHEDULE_CACHE)

	companion object {
		private const val EMAIL_KEY = "EMAIL"
		private const val SCHEDULE_CACHE = "SCHEDULE_CACHE"
	}
}